### ISSUE TITLE
Peacock crashes if OpenGL 3.2 is not supported.

### ISSUE THREAD

**Issue by brianmoose** [idaholab/moose#8800]
On an Ubuntu 16 virtual machine, running `peacock some_exodus.e`
crashes with:
```
ERROR: In /home/algebk/conda-bld/work/VTK-7.1.0/Rendering/OpenGL2/vtkOpenGLRenderWindow.cxx, line 640
vtkXOpenGLRenderWindow (0x442c690): GL version 2.1 with the gpu_shader4 extension is not supported by your graphics driver but is required for the new OpenGL rendering backend. Please update your OpenGL driver. If you are using Mesa please make sure you have version 10.6.5 or later and make sure your driver in Mesa supports OpenGL 3.2.
```

It looks like VTK 7 defaults to requiring OpenGL 2. Apparently there is a possibility of using an older OpenGL backend as well.

At the very least peacock shouldn't crash.
Maybe we should detect this (and possibly Qt5, matplotlib, etc) on peacock startup and refuse to start if everything is not supported.

**Comment by friedmud** [comment]
Peacock really needs to work in a VM... whatever has to happen to make that
possible...

**Comment by brianmoose** [comment]
This was originally reported on the MOOSE users mailing list for an Ubuntu 14.04 machine.
Some opengl info that was produced on that machine:
```
OpenGL vendor string: X.Org
OpenGL renderer string: Gallium 0.4 on AMD RS880
OpenGL core profile version string: 3.1 (Core Profile) Mesa 10.1.3
OpenGL core profile shading language version string: 1.40
OpenGL core profile context flags: (none)
OpenGL core profile extensions:
OpenGL version string: 3.0 Mesa 10.1.3
OpenGL shading language version string: 1.30
OpenGL context flags: (none)
OpenGL extensions:
```
The Mesa that comes with Ubuntu 14.04 doesn't meet the requirements that VTK needs.
Since we supposedly support Ubuntu 14.04, we probably need to investigate using the older VTK OpenGL backend.

**Comment by brianmoose** [comment]
I built VTK with `VTK_RENDERING_BACKEND=OpenGL`  (OpenGL2 is the default) and that allowed to successfully use peacock on a linux Ubuntu 16.04 VM. I am guessing it would also allow it to work on Ubuntu 14.04.

There were quite a few image diffs in the tests on the mac but that is to be expected. The images I looked at didn't look as good as the gold files (which uses OpenGL2).

@aeslaughter Do you see any alternatives to this route?

**Comment by brianmoose** [comment]
I uploaded a linux VTk 7.1.1 built with the OpenGL backend on the conda idaholab channel.
`conda install -c idaholab vtk=7.1.1.opengl` 
I looked at a couple of .e files and saw some strangeness when viewing inside a VM.
`python/chigger/tests/input/simple_diffusion_out.e` has squares not rendered at all on the surface.
`python/chigger/tests/input/diffusion_3.e` the colorbar is completely wrong.
Also, I noticed that the vtkRenderWindow seems to float on top of everything else. So new windows are hidden behind the vtkRenderWindow.
@aeslaughter Suggestions on fixes?

**Comment by aeslaughter** [comment]
Yes, use a newer OpenGL. You can also try to call the resetCamera method on
the chigger.RenderWindow object before it renders/writes.

**Comment by brianmoose** [comment]
It would be nice if things generally worked with both backends. The only difference being speed and quality.

**Comment by aeslaughter** [comment]
Did the resetCamera method do anything to help?

**Comment by brianmoose** [comment]
Well, I **disabled** 3d acceleration in my Ubuntu 16.04 VM and then everything works fine with the stock miniconda-dev environment  (none of the previous issues). I guess with 3d acceleration turned on it uses the OpenGL driver (Chromium) supplied with the VM which isn't very good. Disabling 3d acceleration forces the software rendering using Mesa and everything works fine.

This still doesn't help users on Ubuntu 14.04 where the Mesa version isn't new enough.  They might be able to add a ppa or compile a newer Mesa themselves but that seems like a lot to ask just to run peacock.

**Comment by brianmoose** [comment]
We don't have an easy workaround for Ubuntu 14.
When Ubuntu 18.04 comes out we will drop support for Ubuntu 14.04 so maybe this doesn't matter much?

**Comment by friedmud** [comment]
Fine with me - we have more important things to fix in Peacock anyway...
